src/modules/languages/python.nix

```diff
@@ -135,6 +135,14 @@ in
       description = "Whether `pip install` should avoid outputting messages during devenv initialisation.";
     };
 
+    directory = lib.mkOption {
```

Comment:
does it work if you do `./.` with `types.path` and then do `config.devenv.root + cfg.directory`?

Reply:
I'm don't think I understand what you mean.

In https://github.com/cachix/devenv/pull/993#issuecomment-1973478436 I mentioned that using `types.str` means not copying the specified directory to the Nix store, so poetry is still able to write the lock file.

It also gives users more flexibility when setting the option as they can set it to either an absolute path or a path relative to the devenv root with something like `./directory`. I've added an example to the option.

Reply:
I think you should be still able to use `types.path` to validate it's a path and then do `mystringpath + cfg.directory`